pkg/commands/catalog/mirror/mirror.go

```diff
@@ -508,3 +526,47 @@ func unmarshallObjects(listOfYamls []string) []unstructured.Unstructured {
 	}
 	return toReturn
 }
+
+func downloadArchive(images []string, archivePath string) error {
+	// Create a temporary directory to place the oci-archive files
+	ociArchiveDir, err := os.MkdirTemp("", "oci-archive")
+	if err != nil {
+		return err
+	}
+	defer os.RemoveAll(ociArchiveDir)
+	counter := 1
+	for _, image := range images {
+		imageInfo, err := imageUtil.SplitImage(image)
+		if err != nil {
+			return err
+		}
+		log.Debugf("Copying %s:%s to system", imageInfo.BaseImage, imageInfo.Tag)
+		err = copyOneImage(image, ociArchiveDir, counter, imageInfo)
+		if err != nil {
+			return fmt.Errorf("download failed due to Internal Server Error")
+		}
+		log.Debugf("Successfully pulled image %s out of %s images", strconv.Itoa(counter), strconv.Itoa(len(images)))
+		counter = counter + 1
+	}
+	log.Debugf("Successfully pulled all images. Creating image archives at %s", archivePath)
+	err = dump.CreateReportArchive(ociArchiveDir, archivePath)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func copyOneImage(image string, ociArchiveDir string, counter int, imageInfo types.ImageInfo) error {
+	err := imageUtil.Copy(fmt.Sprintf("docker://%s", image), "oci-archive:"+ociArchiveDir+"/"+strconv.Itoa(counter)+".oci:"+imageInfo.BaseImage+":"+imageInfo.Tag, "", copy.CopyAllImages)
+	if err == nil {
+		return nil
+	} else if !strings.Contains(err.Error(), "500 Internal Server Error") {
+		return err
+	} else {
```

Comment:
get rid of else since you returned above

Reply:
Thanks! I edited this code to include the retry logic and tested that this retry logic works.